### PR TITLE
PR: Add 'Options menu' and a way to move notebooks ('Save as…')

### LINKED ISSUE #13: Create an Options menu with a way to move notebooks
So people could use an untitled notebook to do some exploratory work and then they'd like to rename and move that notebook to another place.

We need to add an option in the Options menu of the notebook to do exactly that :-)

@dalthviz, please work on this one too.

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #13

### CODE REVIEW

File: spyder_notebook/notebookplugin.py [hunk 1]
```diff
@@ -220,6 +232,20 @@ def close_client(self, index=None, client=None):
         self.tabwidget.removeTab(self.tabwidget.indexOf(client))
         self.clients.remove(client)
 
+    def save_as(self):
+        current_client = self.get_current_client()
```
Comment (goanpeca) [inline spyder_notebook/notebookplugin.py]: Docstring :-p

**Comment by ccordoba12** [comment]
Good job @dalthviz! This is exactly what I asked for, but there's one thing I didn't consider: before writing the contents of the untitled file to its new location, we need to ask the server to save it. Else (unless users have pressed the *Save* button in the notebook interface) the contents of untitled will be lost.

For that, you need to do a request to the server URL, using the `requests` package. Every `NotebookClient` widget has a `server_url` and `path` attributes that you need to use for that.

With `requests`, you need to build a query similar to this one:

https://github.com/jupyter/notebook/blob/master/notebook/services/contents/tests/test_contents_api.py#L557

where `save` is defined here:

https://github.com/jupyter/notebook/blob/master/notebook/services/contents/tests/test_contents_api.py#L103

I don't know if this is clear to you. If not, please ask me for more directions :-)

**Comment by dalthviz** [comment]
@ccordoba12 In order to create the request there is a need of a body for the request (in this case a `JSON` with the info from the form). So, I have a question, how I can access the information that the user has at the moment in the form (input) of the notebook (the web page)?.

**Comment by ccordoba12** [comment]
To get the contents of a notebook (after saving it) you can do this request:

```python
import requests
read_req = requests.get('http://localhost:8889/api/contents/untitled0.ipynb')
nb_contents = read_req.content
```

Of course, instead of `http://localhost:8889/`, you need to use `self.server_url`. and create the url request using `url_path_join` :-)

----

For future reference, you can find the full REST API to the notebook server here:

http://petstore.swagger.io/?url=https://raw.githubusercontent.com/jupyter/notebook/master/notebook/services/api/api.yaml

**Comment by ccordoba12** [comment]
This is ready, right?

**Comment by dalthviz** [comment]
Yep 👍

**Comment by ccordoba12** [comment]
Great, thanks!

**Pull request merged**